src/boot/linux.c

```diff
@@ -241,7 +244,8 @@ EFI_STATUS linux_exec(
                                 compat_entry_point,
                                 cmdline,
                                 kernel,
-                                initrd);
+                                initrd,
+                                (EFI_DEVICE_PATH *) kernel_file_path);
```

Comment:
Hmm? The cast looks not trivial.
Maybe `&kernel_file_path->memmap_path.header` is better.

Reply:
IMO this and the next comment are mutually exclusive. I decided to go with the latter